### ISSUE TITLE
Templates should be optional

### ISSUE THREAD

**Issue by lambtho12** [renerocksai/telekasten.nvim#16]
As there are two possibilities to create notes (`new_note()` and `new_templated_note()`), it should be possible to create new notes without having a template set up.

Currently, if `new_note()` is called without an actual template in place, telekasten throws an error as it can not find a template. This is because it defaults to create a templated note when the file does not exist.
https://github.com/renerocksai/telekasten.nvim/blob/a6262253990dcaa08a5b4d8b60e7c244ed86a43f/lua/telekasten.lua#L766-L769

In my opinion, `new_note()` should create a blank note, while `new_tempalted_note()` should create one from the template.

**Comment by renerocksai** [comment]
Thanks for bringing this up. However, I only agree in part with your opinion. Have you tried creating an `empty.md` and using that as template?

If only `new_templated_note()` used templates, I (as in: me) would always have to pick a template, even though it's always the same one for new notes, which I would find annoying.

In addition, following a link to a `[[not-yet existing note]]` would also not have a template - or require an optional template picker that I would find equally annoying since a) I am so used to having my default template filled in, and b) it would require an extra config option: template picker or not.

So, for the sake of sane defaults, I recommend the following:

- not specifying a new note template (setting it to `nil`) should just ignore the templating and create an empty note
- this goes for all types of templates

That way everyone wins :smile: - and it is, in my opinion, the superior solution compared to what we have now.

Will implement soon...

**Comment by lambtho12** [comment]
I agree with you, I did not realize the `new_templated_note` prompted to pick a template. I thought it was always picking the same one. So indeed, I understand the issue with adding an extra step that would be useless in most instances.

Also, I tried setting it to `nil` as well before opening the issue just to be sure :smile: So that was actually my first go-to solution as well.

> Will implement soon...

Thanks. Depending on the "soon" value, it would maybe be nice to add a point in the readme/doc saying that the user is required to create templates (even empty ones) otherwise new notes will not work.
